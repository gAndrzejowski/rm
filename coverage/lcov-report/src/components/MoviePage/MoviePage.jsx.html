<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/MoviePage/MoviePage.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/MoviePage</a> MoviePage.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import styles from './MoviePage.scss';
&nbsp;
export default function MoviePage(props) {
    const {
        title,
        release_date,
        poster_path,
        overview,
        runtime,
        vote_average,
        genres
    } = props.movie;
    return (
    &lt;div className={styles.moviePage}&gt;
        &lt;img className={styles.poster} src={poster_path} /&gt;
        &lt;div className={styles.details}&gt;
            &lt;h1&gt;{title} &lt;span className={styles.rating}&gt;{vote_average}&lt;/span&gt; &lt;/h1&gt;
            &lt;small&gt;{genres.join(', ')}&lt;/small&gt;
            &lt;p&gt;{`${release_date.substring(0,4)}    ${runtime} min`}&lt;/p&gt;
            &lt;p&gt;{overview}&lt;/p&gt;
        &lt;/div&gt;
    &lt;/div&gt;
    );
}
&nbsp;
MoviePage.propTypes = {
   movie: PropTypes.object
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 10 2018 13:08:58 GMT+0200 (Central European Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
