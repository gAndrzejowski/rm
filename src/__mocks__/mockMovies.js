const mockMovies = [
    {
        "id": 185567,
        "title": "Zulu",
        "tagline": "",
        "vote_average": 6.7,
        "vote_count": 200,
        "release_date": "2013-12-04",
        "poster_path": "https://image.tmdb.org/t/p/w500/xg7Dh7mjevDgznqw9JHYccNo9ZQ.jpg",
        "overview": "As a child, Ali Neuman narrowly escaped being murdered by Inkhata, a militant political party at war with Nelson Mandela's African National Congress. Only he and his mother survived the carnage of those years. But as with many survivors, the psychological scars remain.",
        "budget": 16000000,
        "revenue": 0,
        "genres": [
            "Crime",
            "Drama",
            "Thriller"
        ],
        "runtime": 110
    },
    {
        "id": 269149,
        "title": "Zootopia",
        "tagline": "Welcome to the urban jungle.",
        "vote_average": 7.7,
        "vote_count": 6795,
        "release_date": "2016-02-11",
        "poster_path": "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
        "overview": "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
        "budget": 150000000,
        "revenue": 1023784195,
        "genres": [
            "Animation",
            "Adventure",
            "Family",
            "Comedy"
        ],
        "runtime": 108
    },
    {
        "id": 19908,
        "title": "Zombieland",
        "tagline": "This place is so dead",
        "vote_average": 7.2,
        "vote_count": 4509,
        "release_date": "2009-10-07",
        "poster_path": "https://image.tmdb.org/t/p/w500/vUzzDpVrab1BOG3ogxhRGfLN94d.jpg",
        "overview": "Columbus has made a habit of running from what scares him. Tallahassee doesn't have fears. If he did, he'd kick their ever-living ass. In a world overrun by zombies, these two are perfectly evolved survivors. But now, they're about to stare down the most terrifying prospect of all: each other.",
        "budget": 23600000,
        "revenue": 102391382,
        "genres": [
            "Comedy",
            "Horror"
        ],
        "runtime": 88
    },
    {
        "id": 1949,
        "title": "Zodiac",
        "tagline": "There's more than one way to lose your life to a killer.",
        "vote_average": 7.3,
        "vote_count": 2780,
        "release_date": "2007-03-02",
        "poster_path": "https://image.tmdb.org/t/p/w500/bgLyOROfFQI3FqYL7jQbiaV8lkN.jpg",
        "overview": "The true story of the investigation of the \"Zodiac Killer\", a serial killer who terrified the San Francisco Bay Area, taunting police with his ciphers and letters.  The case becomes an obsession for four men as their lives and careers are built and destroyed by the endless trail of clues.",
        "budget": 65000000,
        "revenue": 84785914,
        "genres": [
            "Crime",
            "Drama",
            "Mystery",
            "Thriller"
        ],
        "runtime": 157
    },
    {
        "id": 97630,
        "title": "Zero Dark Thirty",
        "tagline": "The Greatest Manhunt in History",
        "vote_average": 6.7,
        "vote_count": 2028,
        "release_date": "2012-12-19",
        "poster_path": "https://image.tmdb.org/t/p/w500/yg6IDNucLAEj7E5loTyTnUW2pgb.jpg",
        "overview": "A chronicle of the decade-long hunt for al-Qaeda terrorist leader Osama bin Laden after the September 2001 attacks, and his death at the hands of the Navy S.E.A.L. Team 6 in May, 2011.",
        "budget": 40000000,
        "revenue": 132820716,
        "genres": [
            "Thriller",
            "Drama",
            "History"
        ],
        "runtime": 157
    },
    {
        "id": 6795,
        "title": "Zathura: A Space Adventure",
        "tagline": "Adventure Is Waiting",
        "vote_average": 6.1,
        "vote_count": 1080,
        "release_date": "2005-11-06",
        "poster_path": "https://image.tmdb.org/t/p/w500/amqgIuISRBt8tsZM6cTT6gO9WLR.jpg",
        "overview": "After their father is called into work, two young boys, Walter and Danny, are left in the care of their teenage sister, Lisa, and told they must stay inside. Walter and Danny, who anticipate a boring day, are shocked when they begin playing Zathura, a space-themed board game, which they realize has mystical powers when their house is shot into space. With the help of an astronaut, the boys attempt to return home.",
        "budget": 65000000,
        "revenue": 64321501,
        "genres": [
            "Family",
            "Fantasy",
            "Science Fiction",
            "Adventure"
        ],
        "runtime": 101
    },
    {
        "id": 133931,
        "title": "Zambezia",
        "tagline": "Airborne to be wild.",
        "vote_average": 5.2,
        "vote_count": 89,
        "release_date": "2012-06-05",
        "poster_path": "https://image.tmdb.org/t/p/w500/dLZTSTe4czdJ3Fr6vwy8XwVzLF5.jpg",
        "overview": "Set in a bustling bird city on the edge of the majestic Victoria Falls, \"Zambezia\" is the story of Kai - a na√Øve, but high-spirited young falcon who travels to the bird city of \"Zambezia\" where he discovers the truth about his origins and, in defending the city, learns how to be part of a community",
        "budget": 20000000,
        "revenue": 0,
        "genres": [
            "Comedy",
            "Animation",
            "Adventure",
            "Family"
        ],
        "runtime": 83
    },
    {
        "id": 10358,
        "title": "Zack and Miri Make a Porno",
        "tagline": "What would you do to get out of debt?",
        "vote_average": 6.1,
        "vote_count": 894,
        "release_date": "2008-09-18",
        "poster_path": "https://image.tmdb.org/t/p/w500/zxwEJBhSSu5AuKTTd4ugWaCqmg.jpg",
        "overview": "Lifelong platonic friends Zack and Miri look to solve their respective cash-flow problems by making an adult film together. As the cameras roll, however, the duo begin to sense that they may have more feelings for each other than they previously thought.",
        "budget": 24000000,
        "revenue": 42105111,
        "genres": [
            "Comedy"
        ],
        "runtime": 102
    }
]
export default mockMovies;