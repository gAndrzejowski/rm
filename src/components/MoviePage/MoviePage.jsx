// @flow
import React from 'react';
import injectStylesheet from 'react-jss';
import styles from './MoviePage.styles';
import type { MovieData } from '../../flowTypes';

const MoviePage = (props :{movie: MovieData, classes: Object}) => {
  const {
    title,
    release_date: releaseDate,
    poster_path: posterPath,
    overview,
    runtime,
    vote_average: voteAverage,
    genres,
  } = props.movie;
  const { classes } = props;
  return (
    <div className={classes.moviePage}>
      <img className={classes.poster} src={posterPath} alt="" />
      <div className={classes.details}>
        <h1>{title} <span className={classes.rating}>{voteAverage}</span> </h1>
        <small>{genres.join(', ')}</small>
        <p>{`${releaseDate.substring(0, 4)}    ${runtime} min`}</p>
        <p>{overview}</p>
      </div>
    </div>
  );
};
MoviePage.defaultProps = {
  classes: {},
};

export default injectStylesheet(styles)(MoviePage);
