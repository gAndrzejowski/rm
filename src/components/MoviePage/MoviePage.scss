.moviePage {

}
.details,.poster {
  display:inline-block;
}
.details {
  max-width:40vw;
  margin-left:5vw;
}
.poster {
  max-width:30vw;
  max-height:200px;
}
