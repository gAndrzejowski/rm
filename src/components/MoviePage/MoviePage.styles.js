const styles = {
  moviePage: {

  },
  'details, poster': {
    display: 'inline-block',
  },
  details: {
    maxWidth: '40vw',
    marginLeft: '5vw',
  },
  poster: {
    maxWidth: '30vw',
    maxHeight: 200,
  },
  rating: {
    color: 'olivedrab',
    display: 'inline-block',
    borderRadius: '50%',
    border: [1, 'solid', 'olivedrab'],
    padding: '.3em',
  },
};
export default styles;
