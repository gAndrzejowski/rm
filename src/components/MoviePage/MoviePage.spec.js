import React from 'react';
import renderer from 'react-test-renderer';
import MoviePage from './MoviePage';
import mockMovies from '../../__mocks__/mockMovies';

describe('MoviePage component', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<MoviePage movie={mockMovies[0]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
