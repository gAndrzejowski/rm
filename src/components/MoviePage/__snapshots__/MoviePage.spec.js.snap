// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage component renders correctly 1`] = `
<div
  className={undefined}
>
  <img
    className={undefined}
    src="https://image.tmdb.org/t/p/w500/xg7Dh7mjevDgznqw9JHYccNo9ZQ.jpg"
  />
  <div
    className={undefined}
  >
    <h1>
      Zulu
       
      <span
        className={undefined}
      >
        6.7
      </span>
       
    </h1>
    <small>
      Crime, Drama, Thriller
    </small>
    <p>
      2013    110 min
    </p>
    <p>
      As a child, Ali Neuman narrowly escaped being murdered by Inkhata, a militant political party at war with Nelson Mandela's African National Congress. Only he and his mother survived the carnage of those years. But as with many survivors, the psychological scars remain.
    </p>
  </div>
</div>
`;
